docs/user-guide/configmap.md

```diff
@@ -0,0 +1,440 @@
+By default this is disabled.
+
+#### error-log-level
+
+Configures the logging level of errors. Log levels above are listed in the order of increasing severity.
+
+_References:_
+
+- http://nginx.org/en/docs/ngx_core_module.html#error_log
+
+#### gzip-types
+
+Sets the MIME types in addition to "text/html" to compress. The special value "\*" matches any MIME type.
+Responses with the "text/html" type are always compressed if `use-gzip` is enabled.
+
+#### hsts
+
+Enables or disables the header HSTS in servers running SSL.
+HTTP Strict Transport Security (often abbreviated as HSTS) is a security feature (HTTP header) that tell browsers that it should only be communicated with using HTTPS, instead of using HTTP. It provides protection against protocol downgrade attacks and cookie theft.
+
+_References:_
+
+- https://developer.mozilla.org/en-US/docs/Web/Security/HTTP_strict_transport_security
+- https://blog.qualys.com/securitylabs/2016/03/28/the-importance-of-a-proper-http-strict-transport-security-implementation-on-your-web-server
+
+#### hsts-include-subdomains
+
+Enables or disables the use of HSTS in all the subdomains of the server-name.
+
+#### hsts-max-age
+
+Sets the time, in seconds, that the browser should remember that this site is only to be accessed using HTTPS.
+
+#### hsts-preload
+
+Enables or disables the preload attribute in the HSTS feature (when it is enabled)
+
+#### ignore-invalid-headers
+
+Set if header fields with invalid names should be ignored.
+By default this is enabled.
+
+#### keep-alive:
+
+Sets the time during which a keep-alive client connection will stay open on the server side.
+The zero value disables keep-alive client connections.
+
+_References:_
+
+- http://nginx.org/en/docs/http/ngx_http_core_module.html#keepalive_timeout
+
+#### load-balance
+
+Sets the algorithm to use for load balancing.
+The value can either be:
+
+- round_robin: to use the default round robin loadbalancer
+- least_conn: to use the least connected method
+- ip_hash: to use a hash of the server for routing.
+
+The default is least_conn.
+
+_References:_
+
+- http://nginx.org/en/docs/http/load_balancing.html.
+
+#### log-format-upstream
+
+Sets the nginx [log format](http://nginx.org/en/docs/http/ngx_http_log_module.html#log_format).
+Example for json output:
+
+```console
+log-format-upstream: '{ "time": "$time_iso8601", "remote_addr": "$proxy_protocol_addr",
+    "x-forward-for": "$proxy_add_x_forwarded_for", "request_id": "$request_id", "remote_user":
+    "$remote_user", "bytes_sent": $bytes_sent, "request_time": $request_time, "status":
+    $status, "vhost": "$host", "request_proto": "$server_protocol", "path": "$uri",
+    "request_query": "$args", "request_length": $request_length, "duration": $request_time,
+    "method": "$request_method", "http_referrer": "$http_referer", "http_user_agent":
+    "$http_user_agent" }'
+  ```
+
```

Comment:
I would also make a reference to docs/user-guide/log-format.md here, so user can understand what each field represents :)

Reply:
done